dm/master/workerrpc/workerrpc_test.go

```diff
@@ -0,0 +1,164 @@
+	rpcCli, err := NewGRPCClient(addr, config.Security{})
+	c.Assert(err, IsNil)
+
+	// replace the underlying DM-worker client.
+	workerCli := pbmock.NewMockWorkerClient(ctrl)
+	rpcCli.client = workerCli
+
+	reqs := []*Request{
+		{
+			Type:         CmdStartSubTask,
+			StartSubTask: &pb.StartSubTaskRequest{Task: "invalid task content"},
+		},
+		{
+			Type:           CmdOperateSubTask,
+			OperateSubTask: &pb.OperateSubTaskRequest{Op: pb.TaskOp_Pause},
+		},
+		{
+			Type:          CmdUpdateSubTask,
+			UpdateSubTask: &pb.UpdateSubTaskRequest{Task: "another invalid task content"},
+		},
+		{
+			Type:        CmdQueryStatus,
+			QueryStatus: &pb.QueryStatusRequest{Name: "test"},
+		},
+		{
+			Type:       CmdQueryError,
+			QueryError: &pb.QueryErrorRequest{Name: "test-2"},
+		},
+		{
+			Type:              CmdQueryWorkerConfig,
+			QueryWorkerConfig: &pb.QueryWorkerConfigRequest{},
+		},
+		{
+			Type:              CmdSwitchRelayMaster,
+			SwitchRelayMaster: &pb.SwitchRelayMasterRequest{},
+		},
+		{
+			Type:         CmdOperateRelay,
+			OperateRelay: &pb.OperateRelayRequest{Op: pb.RelayOp_ResumeRelay},
+		},
+		{
+			Type:       CmdPurgeRelay,
+			PurgeRelay: &pb.PurgeRelayRequest{Inactive: true},
+		},
+		{
+			Type:        CmdUpdateRelay,
+			UpdateRelay: &pb.UpdateRelayRequest{Content: "invalid relay content"},
+		},
+		{
+			Type:         CmdMigrateRelay,
+			MigrateRelay: &pb.MigrateRelayRequest{BinlogName: "mysql-bin.000123"},
+		},
+		{
+			Type:          CmdOperateSchema,
+			OperateSchema: &pb.OperateWorkerSchemaRequest{Op: pb.SchemaOp_SetSchema},
+		},
+		{
+			Type:          CmdOperateV1Meta,
+			OperateV1Meta: &pb.OperateV1MetaRequest{Op: pb.V1MetaOp_RemoveV1Meta},
+		},
+		{
+			Type:        CmdHandleError,
+			HandleError: &pb.HandleWorkerErrorRequest{Op: pb.ErrorOp_Replace},
+		},
+	}
+
+	workerCli.EXPECT().StartSubTask(gomock.Any(), reqs[0].StartSubTask)
+	workerCli.EXPECT().OperateSubTask(gomock.Any(), reqs[1].OperateSubTask)
+	workerCli.EXPECT().UpdateSubTask(gomock.Any(), reqs[2].UpdateSubTask)
+	workerCli.EXPECT().QueryStatus(gomock.Any(), reqs[3].QueryStatus)
+	workerCli.EXPECT().QueryError(gomock.Any(), reqs[4].QueryError)
+	workerCli.EXPECT().QueryWorkerConfig(gomock.Any(), reqs[5].QueryWorkerConfig)
+	workerCli.EXPECT().SwitchRelayMaster(gomock.Any(), reqs[6].SwitchRelayMaster)
+	workerCli.EXPECT().OperateRelay(gomock.Any(), reqs[7].OperateRelay)
+	workerCli.EXPECT().PurgeRelay(gomock.Any(), reqs[8].PurgeRelay)
+	workerCli.EXPECT().UpdateRelayConfig(gomock.Any(), reqs[9].UpdateRelay)
+	workerCli.EXPECT().MigrateRelay(gomock.Any(), reqs[10].MigrateRelay)
+	workerCli.EXPECT().OperateSchema(gomock.Any(), reqs[11].OperateSchema)
+	workerCli.EXPECT().OperateV1Meta(gomock.Any(), reqs[12].OperateV1Meta)
+	workerCli.EXPECT().HandleError(gomock.Any(), reqs[13].HandleError)
+
+	for _, req := range reqs {
```

Comment:
could be combined to below loop

Reply:
done in https://github.com/pingcap/dm/pull/1047/commits/d919dc5317f620f5287190df35a1bd3823e08759.